### ISSUE TITLE
Accuracy is stuck 

### ISSUE THREAD

**Issue by shreejalt** [yizhe-ang/AdaMatch-PyTorch#1]
Hi @yizhe-ang 
Thanks for your contribution. 
I tested your code in dassl.pytorch and ran it. But the accuracy is stuck at 20 after 30 epochs. Is it normal to see that behavior?

Also, can you share your reproduced result tensorboard logs so that I can make a comparison?

**Comment by yizhe-ang** [comment]
Hi @shreejalt, can I check which experiment / dataset are you running?

**Comment by shreejalt** [comment]
hi @yizhe-ang 
It took time to converge when I tested it. But it happened.

I wanted to ask you that your code is UDA. If I want to convert it into SSDA then I just have to concatenate the labeled target with the labeled source and do the same flow of the algorithm?

Also, in the distribution alignment they take the ratio of labeled source / unlabeled target. In my case will it be (labeled target) / unlabeled target ???

Because in the paper it is mentioned that if the target distribution is known then we can replace the labeled source with labeled target..

`If the target label distribution is known, one can simply replace the term E[Yˆ
SL,w] with it in the formula above.`

Can you please help me with this?? 



P.S In their JAX codebase of adamatch, they do (labeled source / unlabeled target) in SSDA in distribution alignment. Isnt it wrong? Because if we have target labeled examples in SSDA should we not divide the unlabeled target with the labeled target?

**Comment by shreejalt** [comment]
Also, why are you doinf `self.update_lr()` after iteration? Shouldnt it be after every epoch? Correct me if I am wrong

**Comment by shreejalt** [comment]
I just trained it again... the accuracy dips after some 500 epochs
![Screen Shot 2022-01-23 at 10 57 33 PM](https://user-images.githubusercontent.com/22684229/150690411-f0ce677c-6b87-46d9-9f8b-e5774afee9b7.png)

**Comment by yizhe-ang** [comment]
Apologies for the late response.

`self.update_lr()` is done after every iteration as it's using a custom cosine schedule (as per the paper and the official implementation).

With regard to your reported experiments, from my experience the algorithm doesn't seem to be very robust and achieves varying performances with different seeds (this could be due to a mistake in my implementation).

If you are running this on your own application, perhaps you could try some form of early stopping.
